test/ui/login/login_view_model_test.dart

```diff
@@ -0,0 +1,116 @@
+import 'package:flutter_survey/ui/login/login_screen.dart';
+import 'package:flutter_survey/ui/login/login_state.dart';
+import 'package:flutter_survey/ui/login/login_view_model.dart';
+import 'package:flutter_survey/database/secure_storage.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/mockito.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:flutter_survey/api/exception/network_exceptions.dart';
+import 'package:flutter_survey/usecases/base/base_use_case.dart';
+import 'package:flutter_survey/model/login_model.dart';
+
+import '../../mocks/generate_mocks.mocks.dart';
+
+void main() {
+  group(
+    'LoginViewModelTest',
+    () {
+      late LoginViewModel loginViewModel;
+      late MockSecureStorage secureStorage;
+      late MockLogInUseCase mockLogInUseCase;
+      late MockAuthTokenStorageUseCase mockAuthTokenStorageUseCase;
+      late ProviderContainer providerContainer;
+      const loginModel = LoginModel(
+        accessToken: "accessToken",
+        tokenType: "tokenType",
+        expiresIn: 10,
+        refreshToken: "refreshToken",
+        createdAt: 1,
+      );
+      setUp(
+        () {
+          mockLogInUseCase = MockLogInUseCase();
+          mockAuthTokenStorageUseCase = MockAuthTokenStorageUseCase();
+          secureStorage = MockSecureStorage();
+          loginViewModel = LoginViewModel(
+            mockLogInUseCase,
+            mockAuthTokenStorageUseCase,
+          );
+          providerContainer = ProviderContainer(
+            overrides: [
+              loginViewModelProvider.overrideWithValue(loginViewModel),
+            ],
+          );
+          loginViewModel =
+              providerContainer.read(loginViewModelProvider.notifier);
+        },
+      );
+      test(
+        'When starting the app init state will appear',
+        () {
+          expect(
+            providerContainer.read(loginViewModelProvider),
+            const LoginState.init(),
+          );
+        },
+      );
+      test(
+        'When credential provided in correct format, login success occurred and token stored',
```

Comment:
We don't care about the validation in the VM layer, should we mention about the `credential provided in correct format` here?

```suggestion
        'When logging in & storing the token successfully, it emits loginSuccess state to navigate to the Home screen',
```